lib/token.js

```diff
@@ -379,26 +379,44 @@ function getIdToken(sdk, oauthOptions, options) {
   function handleOAuthResponse(res) {
     if (res['error'] || res['error_description']) {
       throw new OAuthError(res['error'], res['error_description']);
+    }
 
-    } else if (res['id_token']) {
-      if (res.state !== oauthParams.state) {
-        throw new AuthSdkError('OAuth flow response state doesn\'t match request state');
-      }
+    if (res.state !== oauthParams.state) {
+      throw new AuthSdkError('OAuth flow response state doesn\'t match request state');
+    }
+    
+    var constructedToken = {
+      scopes: oauthQueryHash.scope.split(' ')
+    };
 
+    if (res['id_token']) {
       var jwt = sdk.idToken.decode(res['id_token']);
       if (jwt.payload.nonce !== oauthParams.nonce) {
         throw new AuthSdkError('OAuth flow response nonce doesn\'t match request nonce');
       }
 
       validateClaims(jwt.payload, sdk.options.url, oauthParams.clientId);
-      return {
+
+      util.extend(constructedToken, {
         idToken: res['id_token'],
-        claims: jwt.payload
-      };
+        claims: jwt.payload,
+        expiresAt: jwt.payload.exp
+      });
+    }
+    
+    if (res['access_token']) {
+      util.extend(constructedToken, {
+        accessToken: res['access_token'],
+        expiresAt: Number(res['expires_in']) + Math.floor(Date.now()/1000),
```

Comment:
Am curious about the behavior here when asking for id_token and token - do we expect to have a payload.exp for the idToken? And if so, are we overwriting it here with the expiresIn calculation?